core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -2598,13 +2603,23 @@ private Type coerceToSingleType(StackableAstVisitorContext<Context> context, Str
 
         private void addOrReplaceExpressionCoercion(Expression expression, Type type, Type superType)
         {
-            NodeRef<Expression> ref = NodeRef.of(expression);
-            expressionCoercions.put(ref, superType);
-            if (typeCoercion.isTypeOnlyCoercion(type, superType)) {
-                typeOnlyCoercions.add(ref);
+            addOrReplaceExpressionCoercion(expression, superType, typeCoercion.isTypeOnlyCoercion(type, superType));
+        }
+
+        private void addOrReplaceExpressionCoercion(Expression expression, Type superType, boolean isTypeOnlyCoercion)
+        {
+            addOrReplaceExpressionsCoercion(List.of(expression), superType, isTypeOnlyCoercion);
+        }
+
+        private void addOrReplaceExpressionsCoercion(List<Expression> expressions, Type superType, boolean isTypeOnlyCoercion)
```

Comment:
you don't need to pass `boolean isTypeOnlyCoercion` as argument since this method is already batching expressions (no perf gain)

Reply:
Done